src/compiler/checker.ts

```diff
@@ -24924,16 +24937,102 @@ namespace ts {
             }
         }
 
+        function getIntersectedSignatures(signatures: readonly Signature[]) {
+            return !getStrictOptionValue(compilerOptions, "noImplicitAny") ? undefined : reduceLeft(signatures, (left, right) => left === right || !left ? left : (!!left.typeParameters && !!right.typeParameters && compareTypeParametersIdentical(left.typeParameters, right.typeParameters)) || (!left.typeParameters && !right.typeParameters) ? combineSignaturesOfIntersectionMembers(left, right) : undefined, signatures[0]);

```

Comment:
```suggestion
            return getStrictOptionValue(compilerOptions, "noImplicitAny")
                : reduceLeft(
                    signatures,
                    (left, right) =>
                        left === right || !left ? left
                        : compareTypeParametersIdentical(left.typeParameters, right.typeParameters) ? combineSignaturesOfIntersectionMembers(left, right)
                        : undefined)
                ? undefined;
```

1. rely on reduceLeft's `initial` behaviour to do the right thing
2. invert getStrictOptionValue
3. Simplify call to compareTypeParametersIdentical by pushing some code in there.

Note that this relies on pushing the both-or-neither-defined check down into compareTypeParametersIdentical